src/GafferUI/RotateHandle.cpp

```diff
@@ -136,17 +138,10 @@ Imath::Eulerf RotateHandle::rotation( const DragDropEvent &event )
 {
 	if( m_axes == Style::XYZ )
 	{
-		const LineSegment3f line = event.line * fullTransform() * m_dragBeginWorldTransform.inverse();
+		const LineSegment3f line = updatedLineFromEvent( event ) * fullTransform() * m_dragBeginWorldTransform.inverse();
```

Comment:
I haven't sussed out where this is going wrong, but engaging precise mode halfway through a drag is causing a snap back rather than a precise continuation of the current drag.

Reply:
Should be fixed in https://github.com/GafferHQ/gaffer/pull/3438/commits/7197589586a69e94ae18454380437bacd31fa5b8.

Reply:
Also fixed a bug in `shift`-at-start free rotations in https://github.com/GafferHQ/gaffer/pull/3438/commits/1358a01c519374029f6ebf5a6df292782db27df3.